tests/loader/check_loader.py

```diff
@@ -96,7 +96,20 @@ def check_test_loader_with_class(self):
         tests = loader.discover([os.path.join(discover_dir(), "test_b.py")])
         assert len(tests) == 3
 
+    def check_test_loader_with_injected_args(self):
+        """When the --injected command-line option is used, the loader behaves a little bit differently:
 
+        each test method annotated with @parametrize or @matrix should only expand to a single discovered test,
+        and the injected args should be those passed in from command-line.
+        """
+        injected = {"x": 1, "y": -1}
+        loader = TestLoader(self.SESSION_CONTEXT, injected_args=injected)
 
+        file = os.path.join(discover_dir(), "test_decorated.py")
+        tests = loader.discover([file])
+        assert len(tests) == 4
```

Comment:
Tests that take different parameters or none will result cause exceptions, and I think this is the right behavior.

Hmm I suppose you could raise an error if the loader finds more than one test, but I don't really see why it should have that restriction? It may be an unusual use-case to target more than one test this way, but I don't see why it should be an error condition.